From: Dilshod Mukhtarov <dilshodm@gmail.com>
Date: Wed, 29 May 2024 13:10:00 +0400
Subject: [PATCH] Fix avfoundation Mac build error on c++17

avfcamerautility.mm:104:35: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>

diff -Nru a/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm b/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm
--- a/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm	2024-05-08 09:14:49.000000000 +0400
+++ b/qtmultimedia/src/plugins/avfoundation/camera/avfcamerautility.mm	2024-05-29 13:06:51.257069776 +0400
@@ -101,7 +101,7 @@
     }
 };
 
-struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>
+struct FormatHasNoFPSRange : std::__unary_function<AVCaptureDeviceFormat *, bool>
 {
     bool operator() (AVCaptureDeviceFormat *format)
     {
